lib/org/chromium/webidl/Diff.js

```diff
@@ -0,0 +1,247 @@
+// Copyright 2017 The Chromium Authors. ALl rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+'use strict';
+
+foam.CLASS({
+  package: 'org.chromium.webidl',
+  name: 'Diff',
+
+  requires: [
+    'org.chromium.webidl.DiffChunk',
+    'org.chromium.webidl.DiffStatus',
+  ],
+
+  documentation: `The diff algorithm that is performed on canonicalized
+    definitions.`,
+
+  properties: [
+    {
+      class: 'FObjectArray',
+      of: 'org.chromium.webidl.DiffChunk',
+      name: 'chunks',
+    },
+    {
+      name: 'excludedProps',
+      value: [ 'id', 'isCanonical', 'sources', 'source' ],
+    },
+  ],
+
+  methods: [
+    {
+      documentation: `diff expects to receive two maps of canonicalized
```

Comment:
Just plain JavaScript objects. `CanonicalCollection`s are received by `DiffRunner` and only the `definitions` field (of type `Object`) will be passed to `Diff`.